pkg/environment/environment.go

```diff
@@ -0,0 +1,61 @@
+package environment
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+)
+
+// Environment encodes the operating environment for the model runner.
+type Environment uint8
+
+const (
+	// EnvironmentUnknown represents an unknown environment in which basic,
+	// non-specialized defaults should be used.
+	EnvironmentUnknown Environment = iota
+	// EnvironmentDesktop represents a Docker Desktop environment.
+	EnvironmentDesktop
+	// EnvironmentMoby represents a Moby engine environment, if installed via
+	// the model CLI.
+	EnvironmentMoby
+	// EnvironmentCloud represents a Docker Cloud environment, if installed via
+	// the model CLI.
+	EnvironmentCloud
+)
+
+// environment is the cached environment.
+var environment Environment
+
+// environmentOnce guards initialization of environment.
+var environmentOnce sync.Once
+
+// isDockerBackend checks if an executable path is com.docker.backend.
+func isDockerBackend(path string) bool {
+	_, leaf := filepath.Split(path)
```

Comment:
nit:
```suggestion
	leaf := filepath.Base(path)
```
I'd also go with another variable name like `filename` or `execName` instead of `leaf`.